Dockerfile

```diff
@@ -1,11 +1,35 @@
-FROM golang:alpine
+FROM golang:alpine AS builder
 
-RUN mkdir /app
-ADD . /app/
-WORKDIR /app
+# Set necessary environment variables needed for our image
+ENV GO111MODULE=on \
+    CGO_ENABLED=0 \
+    GOOS=linux \
+    GOARCH=amd64
 
-RUN apk add make
+# Move to working directory /build
+WORKDIR /build
 
-RUN make build
+# Copy and download dependency using go mod
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
 
-CMD ["./vedran-daemon"]
+# Copy the code into the container
+COPY . .
+
+# Build the application
+RUN go build -o vedran-daemon .
+
+# Move to /dist directory as the place for resulting binary folder
+WORKDIR /dist
+
+# Copy binary from build to main folder
+RUN cp /build .
```

Comment:
I think this will make the same problem, so I think that the best solution is
`RUN cp /build/vedran-daemon .`